doc/cug.tex

```diff
@@ -4551,6 +4489,380 @@ \subsubsection{Collapsible Families In Suite Graphs}
+(See~\ref{RefParameterTemplates} for more on the string template syntax.)
+
+Any number of parameters can be used at once.  This parameterization,
+\begin{lstlisting}
+    model<run,obs>  # for run = 1..2 and obs = ship, buoy, plane
+\end{lstlisting}
+expands to these tasks names,
+\begin{lstlisting}
+    model_run1_ship, model_run1_buoy, model_run1_plane,
+    model_run2_ship, model_run2_buoy, model_run2_plane
+\end{lstlisting}
+
+\subsubsection{Passing Parameter Values To Tasks}
+
+Parameter values are passed as environment variables to tasks generated by
+parameter expansion.  For the example above, task \lstinline=model_run2_ship=
+would get the following environment variables:
+\begin{lstlisting}
+    $CYLC_TASK_PARAM_run  # value "2"
+    $CYLC_TASK_PARAM_obs  # value "ship"
+\end{lstlisting}
+These variables allow tasks to determine which member of a parameterized
+group they are, and so to vary their behaviour accordingly.
+
+\subsubsection{Parameterized Tasks In The Graph}
+
+Using a different number of parameters on each side of an arrow in the graph
+will branch or join a workflow.  For example this,
+\begin{lstlisting}
+    graph = "prep => model<run>"  # for run = 1, 2
+\end{lstlisting}
+expands to this,
+\begin{lstlisting}
+    prep => model_run1
+    prep => model_run2
+      # or equivalently:
+    prep => model_run1 & model_run2
+\end{lstlisting}
+and this,
+\begin{lstlisting}
+    # for run = 1, 2 and mem = cat, dog
+    graph = """prep => init<run> => model<run,mem> =>
+                    post<run,mem> => wrap<run> => done"""
+\end{lstlisting}
+expands to this,
+\begin{lstlisting}
+    prep => init_run1 => model_run1_cat => post_run1_cat => wrap_run1 => done
+            init_run1 => model_run1_dog => post_run2_dog => wrap_run1
+    prep => init_run2 => model_run2_cat => post_run2_cat => wrap_run2 => done
+            init_run2 => model_run2_dog => post_run2_dog => wrap_run2
+\end{lstlisting}
+This is visualized by \lstinline=cylc graph= in Figure~\ref{fig-params-1}.
+
+\begin{figure}
+    \begin{center}
+        \includegraphics[width=10cm]{graphics/png/orig/params1.png}
+    \end{center}
+    \caption[Parameter expansion example.]{\scriptsize
+     Parameter expansion example.}
+    \label{fig-params-1}
+\end{figure}
+
+\subsubsection{Singling Out Specific Parameter Values}
+
+Specific parameter values can be singled out in the graph and under
+\lstinline=[runtime]= with this notation: \lstinline@<param_name=param_value>@.
+For example, here's how to make a special task trigger off just the first of a
+set of model runs,
+\begin{lstlisting}
+[cylc]
+    [[parameters]]
+        run = 1..5
+[scheduling]
+    [[dependencies]]
+        graph = """model<run> => post_proc<run>  # general case
+                   model<run=1> => check_first_run  # special case"""
+[runtime]
+    [[model<run>]]
+        # config for all "model" runs...
+    [[model<run=1>
+        # special config for the first model run (if any)...
```

Comment:
It's override by repetition (not inheritance): the first heading configures all `model<run>` including for `run=1`, the second one just (re)configures `model<run=1>`.